### PR TITLE
Add more file metadata validation tests

### PR DESCRIPTION
**Pull request by cshannon**
This addresses part of #3766 and adds more tests to verify file metadata validation works properly for columns that used StoredTabletFile metadata. Tests include verifying old format is invalid, the json format is correctly validated and rows are a row range

Original comment: https://github.com/apache/accumulo/pull/3761#discussion_r1334838096

### CODE REVIEW

File: server/base/src/test/java/org/apache/accumulo/server/constraints/MetadataConstraintsTest.java [hunk 1]
```diff
@@ -381,6 +384,41 @@ public void testBulkFileCheck() {
...
+    // Bad Json - only path (old format) so should fail parsing
+    m = new Mutation(new Text("0;foo"));
+    m.put(BulkFileColumnFamily.NAME, new Text("hdfs://1.2.3.4/accumulo/tables/2a/t-0003/someFile"),
+        new Value("5"));
+    violations = mc.check(createEnv(), m);
+    assertNotNull(violations);
+    assertEquals(1, violations.size());
+    assertEquals(Short.valueOf((short) 9), violations.get(0));
+    assertNotNull(mc.getViolationDescription(violations.get(0)));
+
```
Comment (keith-turner) [inline server/base/src/test/java/org/apache/accumulo/server/constraints/MetadataConstraintsTest.java:396]: I was experimenting with these test and replacing startRow with empty string produces json that looks like the following.

```json
{"path":"hdfs://1.2.3.4/accumulo/tables/2a/t-0003/someFile","":"","endRow":""}
```

In the json above the startRow field is missing and an unexpected field with the empty string for a name exists. Thinking it would be nice to add another test case where startRow is missing and there is not an unexpected field.  Did that below, did not attempt a regex to create the json though.

```suggestion
    // Bad Json - test startRow key missing so validation should fail
    m = new Mutation(new Text("0;foo"));
    m.put(BulkFileColumnFamily.NAME, new Text("{\"path\":\"hdfs://1.2.3.4/accumulo/tables/2a/t-0003/someFile\",\"endRow\":\"\"}"),
            new Value("5"));
    violations = mc.check(createEnv(), m);
    assertNotNull(violations);
    assertEquals(1, violations.size());
    assertEquals(Short.valueOf((short) 9), violations.get(0));
    assertNotNull(mc.getViolationDescription(violations.get(0)));
```

Looking into this I was wondering if empty string is a valid field name in json, found discussion about this and it appears it is.

**Review by keith-turner — APPROVED** [review]
I did not realize the bulk loaded column was using the new json format.  It will never have the range set, but I guess it uses the json for consistency?  That seems fine, just curious if there is more to it than consistency.

File: server/base/src/test/java/org/apache/accumulo/server/constraints/MetadataConstraintsTest.java (hunk 1 above)
Comment (cshannon) [reply server/base/src/test/java/org/apache/accumulo/server/constraints/MetadataConstraintsTest.java:396]: Good idea, I will work on these and have an update shortly.

**Comment by cshannon** [comment]
@keith-turner - I updated this with some more tests (empty strings in json and missing path) and I did a little cleanup with a helper method for checking violations. Lastly, I added some null checks on the deserialized StoredTabletFile json just to give a better error message then the default "NPE" that happens otherwise which may help someone track down what is actually null (this could be useful if someone manually modifies a file entry and makes a mistake)

**Review by Manno15 — APPROVED** [review]
Recent changes and cleanup look good.
